lib/rspec/core/option_parser.rb

```diff
@@ -97,15 +107,27 @@ def parser(options)
         end
 
         parser.on('-X', '--[no-]drb', 'Run examples via DRb.') do |o|
-          options[:drb] = o
+          if o
+            options[:runner] = -> (opts, err, out) do
+              require 'rspec/core/drb'
+              begin
+                DRbRunner.new(opts).run(err, out)
+                exit
+              rescue DRb::DRbConnError
+                err.puts "No DRb server is running. Running in local process instead ..."
+              end
+            end
+          end
         end
 
         parser.on('--drb-port PORT', 'Port to connect to the DRb server.') do |o|
           options[:drb_port] = o.to_i
         end
 
         parser.on('--init', 'Initialize your project with RSpec.') do |_cmd|
-          initialize_project_and_exit
+          options[:runner] = -> (*_args) do
+            initialize_project_and_exit
+          end
```

Comment:
Could this be `options[:runner] = method(:initialize_project_and_exit)`? The method would have to take the args, but I think that's ok.